modules/tensor_mechanics/src/materials/ComputeEigenstrainFromInitialStress.C

```diff
@@ -0,0 +1,72 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+#include "ComputeEigenstrainFromInitialStress.h"
+#include "RankTwoTensor.h"
+#include "Function.h"
+
+template <>
+InputParameters
+validParams<ComputeEigenstrainFromInitialStress>()
+{
+  InputParameters params = validParams<ComputeEigenstrainBase>();
+  params.addClassDescription("Computes an eigenstrain from an initial stress");
+  params.addRequiredParam<std::vector<FunctionName>>(
+      "initial_stress",
+      "A list of functions describing the initial stress.  If provided, there "
+      "must be 9 of these, corresponding to the xx, yx, zx, xy, yy, zy, xz, yz, "
+      "zz components respectively.  To compute the eigenstrain correctly, your "
+      "elasticity tensor should not be time-varying in the first timestep");
+  params.addParam<std::string>("base_name",
+                               "The base_name for the elasticity tensor that will be "
+                               "used to compute strain from stress.  Do not provide "
+                               "any base_name if your elasticity tensor does not use "
+                               "one.");
+  params.set<bool>("incremental_form") = true;
+  return params;
+}
+
+ComputeEigenstrainFromInitialStress::ComputeEigenstrainFromInitialStress(
+    const InputParameters & parameters)
+  : ComputeEigenstrainBase(parameters),
+    _base_name(isParamValid("base_name") ? getParam<std::string>("base_name") + "_" : ""),
+    _elasticity_tensor(getMaterialPropertyByName<RankFourTensor>(_base_name + "elasticity_tensor"))
+{
+  if (_incremental_form == false)
```

Comment:
I've been wanting to get rid of `_incremental_form` for some time.  There are only a couple of models that use it. I think this is the right thing to do for now, but when we remove `_incremental_form`, we'll just have to have this model get the old eigenstrain on its own.